src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
+/* eslint-disable prettier/prettier */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-sequences */
+/* eslint-disable prettier/prettier */
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    (this.name = name), (this.health = health), Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name), (this.hidden = false);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(obj) {
+    if (obj instanceof Herbivore && !obj.hidden) {
+      obj.health -= 50;
+      obj.checkHealth();
```

Comment:
After decreasing the health of the herbivore, you should filter the `Animal.alive` array to remove any animals with health less than or equal to 0. This can be done directly here instead of calling `checkHealth`.